### ISSUE TITLE
Preflight OPTIONS request requires Authorization header to be set

### ISSUE THREAD

**Issue by JonasR-** [iotaledger/iri#348]
Hi,

I am not really in the thematic but as I read its standardized that a OPTIONS request is send if there are not standard header in a request. This should ensure that all set header are accepted by the server side.

So if I do a POST request with plain javascript (not the IOTA library) with set "Authorization" header, I get a "401" HTTP error. This is because my browser sends a OPTIONS request before the POST request

> Accept:*/*
> Accept-Encoding:gzip, deflate
> Accept-Language:de-DE,de;q=0.8,en-US;q=0.6,en;q=0.4,it;q=0.2
> Access-Control-Request-Headers:authorization,x-iota-api-version
> Access-Control-Request-Method:POST
> Cache-Control:no-cache
> Connection:keep-alive
> DNT:1
> Host: yyy
> Origin: xxx
> Pragma:no-cache
> Referer: xxx
> User-Agent: xxx

As you can see the "Authorization" header is not set its only mentioned in the "Access-Control-Request-Headers". 

Long story short request: Please dont request the "Authorization" on OPTIONS requests

**Comment by JonasR-** [comment]
Minimal example

```javascript
var request = new XMLHttpRequest();
request.onload = function () {
   var status = request.status;
   console.log("status: " + status)
   var data = request.responseText;
   console.log("data: " + data)
}

request.open("POST", "http://NODE_ADDRESS:14265", true);

request.setRequestHeader("Content-Type", "application/json;charset=UTF-8");
request.setRequestHeader("X-IOTA-API-Version", "asd");
request.setRequestHeader("Authentication", "iotatoken:iota");
request.send('{"command": "getNeighbors"}');
```

**Comment by JonasR-** [comment]
Whats the status? ...

**Comment by alon-e** [comment]
>Please dont request the "Authorization" on OPTIONS requests

I didn't find the related commit - but looking at IRI API OPTIONS response code - https://github.com/iotaledger/iri/blob/dev/src/main/java/com/iota/iri/service/API.java#L133
and testing via Chrome dev console I don't see any request for "Authorization":

```Request URL:http://localhost:14265/
Request Method:OPTIONS
Status Code:200 OK
Remote Address:127.0.0.1:14265
Referrer Policy:no-referrer-when-downgrade

Response Headers
Access-Control-Allow-Headers:Origin, X-Requested-With, Content-Type, Accept, X-IOTA-API-Version
Access-Control-Allow-Origin:*
Allow:GET,HEAD,POST,PUT,DELETE,TRACE,OPTIONS,CONNECT,PATCH
Connection:keep-alive
Content-Length:0
Content-Type:text/plain
Date:Sun, 29 Apr 2018 08:22:28 GMT```
